engine/classes/Elgg/Database/Plugins.php

```diff
@@ -398,46 +395,46 @@ function find($status = 'active') {
 			return [];
 		}
 
-		$db_prefix = _elgg_config()->dbprefix;
 		$priority = $this->namespacePrivateSetting('internal', 'priority');
 		$site_guid = 1;
-		
+
 		// grab plugins
 		$options = [
 			'type' => 'object',
 			'subtype' => 'plugin',
-			'limit' => ELGG_ENTITIES_NO_VALUE,
+			'limit' => 0,
 			'selects' => ['ps.value'],
-			'joins' => [
-				"JOIN {$db_prefix}private_settings ps on ps.entity_guid = e.guid",
-			],
-			'wheres' => ["ps.name = '$priority'"],
+			'private_setting_names' => [$priority],
 			// ORDER BY CAST(ps.value) is super slow. We usort() below.
 			'order_by' => false,
 		];
-	
+
 		switch ($status) {
 			case 'active':
 				$options['relationship'] = 'active_plugin';
 				$options['relationship_guid'] = $site_guid;
 				$options['inverse_relationship'] = true;
 				break;
-	
+
 			case 'inactive':
-				$options['wheres'][] = "NOT EXISTS (
-						SELECT 1 FROM {$db_prefix}entity_relationships active_er
-						WHERE active_er.guid_one = e.guid
-							AND active_er.relationship = 'active_plugin'
-							AND active_er.guid_two = $site_guid)";
+				$options['wheres'][] = function (QueryBuilder $qb) use ($site_guid) {
+					$subquery = $qb->subquery('entity_relationships', 'active_er');
+					$subquery->select('*')
+						->where($qb->compare('active_er.guid_one', '=', 'e.guid'))
+						->andWhere($qb->compare('active_er.relationship', '=', 'active_plugin', ELGG_VALUE_STRING))
+						->andWhere($qb->compare('active_er.guid_two', '=', 1));
+
+					return "NOT EXISTS ({$subquery->getSQL()})";
+				};
 				break;
-	
+
 			case 'all':
 			default:
 				break;
 		}
-	
+
 		$old_ia = elgg_set_ignore_access(true);
-		$plugins = elgg_get_entities_from_relationship($options);
+		$plugins = elgg_get_entities($options) ? : [];
```

Comment:
It doesn't always. In case of malformatted data, it returns `false`. `Entities::get` will always return an array, but to keep things BC, `Entities::find` returns false.

Reply:
I can change the behavior of `Entities:;find` to see if all tests pass, but legacy integration tests do `assertFalse` on results of `elgg_get_entities()` sometimes, so that's when it fails I think